File: semestr 5/ON/List2/zad2/zad2.py
# Jędrzej Sajnóg 279701


import matplotlib.pyplot as plt
from math import exp, log
import numpy as np

def f(x):
    return exp(x) * log(1 + exp(-x))



# X = range(0,50,0.01)
X = np.arange(0,50,0.01)
Y = [f(x) for x in X]

plt.plot(X,Y)
plt.savefig("python.png")
plt.show()

File: semestr 5/ON/List4/zad2/zad2.jl
# Jędrzej Sajnóg 279701

function warNewton(x::Vector{Float64}, fx::Vector{Float64}, t::Float64)
    product = fx[length(fx)]
    for i in (length(fx)-1):-1:1
        product = product * (t - x[i]) + fx[i]
    end
    return product
end